Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -40,27 +58,58 @@ public class CameraAnimator: NSObject {
     // MARK: Initializer
     internal init(delegate: CameraAnimatorDelegate,
                   propertyAnimator: UIViewPropertyAnimator,
-                  owner: AnimationOwner) {
+                  owner: AnimationOwner,
+                  cameraView: CameraView = CameraView()) {
         self.delegate = delegate
         self.propertyAnimator = propertyAnimator
         self.owner = owner
+
+        // Set up the short lived camera view
+        self.cameraView = cameraView
+        delegate.addViewToViewHeirarchy(cameraView)
     }
 
     deinit {
         propertyAnimator.stopAnimation(false)
         propertyAnimator.finishAnimation(at: .current)
+        cameraView.removeFromSuperview()
     }
 
-    // MARK: Functions
-
-    /// Starts the animation.
+    /// Starts the animation if this animator is in `inactive` state. Also used to resume a "paused" animation.
     public func startAnimation() {
+
+        if self.state != .active {
+
+            guard let delegate = delegate else {
+                fatalError("CameraAnimator delegate cannot be nil when starting an animation")
+            }
+
+            guard let animation = animation else {
```

Comment:
Won't this always fail since we only set the animation later when `addAnimations` is called?

Reply:
Not sure I follow.. We have a requirement that `addAnimations(...)` is called before `startAnimation()` is called.

Reply:
Nevermind. I was confused.